src/server/pfs/server/driver.go

```diff
@@ -804,59 +804,35 @@ func (d *driver) finishCommit(txnCtx *txnenv.TransactionContext, commit *pfs.Com
 	if description != "" {
 		commitInfo.Description = description
 	}
-	commitPath := commitKey(commit)
-	// Run compaction task.
-	return d.compactionQueue.RunTaskBlock(txnCtx.Client.Ctx(), func(m *work.Master) error {
-		exists := func(p string) (bool, error) {
-			var exists bool
-			if err := d.storage.Store().Walk(m.Ctx(), p, func(_ string) error {
-				exists = true
-				return nil
-			}); err != nil {
-				return false, err
-			}
-			return exists, nil
+	var parentFSID *fileset.ID
+	if commitInfo.ParentCommit != nil {
```

Comment:
Why don't we just do this in the task callback, then just append to the ids when there is a parent?

Reply:
I didn't want to do another call to postgres in the callback.  It seems like the callback that gets the master is contended?